### PR TITLE
Add contao:mailer:send command

### PR DESCRIPTION
**Pull request by fritzmg**
A `mailer:send` command will be added to Symfony only in 6.1 (or higher), if at all (see https://github.com/symfony/symfony/pull/43687). In order to also have a console command for sending emails for our last `4.x` LTS, we should integrate the command into the `contao/core-bundle`. Later on we can deprecate and forward it to the Symfony command again.

The command largely works the same way as the `swiftmailer:email:send` command previously.

### CODE REVIEW

**Review by Toflar — COMMENTED** [review]
👍 on merging this, however I would rename the command to `contao:mailer:send`. It allows for possible new commands in the future and does not imply any connection to the `sendmail` binary.

File: core-bundle/src/Command/SendMailCommand.php
Comment (94noni) [inline core-bundle/src/Command/SendMailCommand.php]: you can leverage https://symfony.com/blog/new-in-symfony-5-3-lazy-command-description

**Review by Toflar — APPROVED** [review]

**Review by bytehead — APPROVED** [review]

**Review by leofeyer — CHANGES_REQUESTED** [review]
I would like to wait and see if Symfony really merges the PR. Depending on that, we should consider providing an extension that backports the functionality to Contao.

**Comment by aschempp** [comment]
> I would like to wait and see if Symfony really merges the PR. Depending on that, we should consider providing an extension that backports the functionality to Contao.

I agree with this. We should add a command with the same name and characteristic (if possible). Which makes me wonder if this should not be in Contao itself but a bundle that can be included if you want to command-line-send in Contao 4.13 🤷

**Comment by fritzmg** [comment]
> I would like to wait and see if Symfony really merges the PR.

Symfony moved it to the `6.1` milestone. We could in theory allow the installation of `symfony/mailer: ^5.4 || ^6.1` though.

**Comment by fritzmg** [comment]
> Depending on that, we should consider providing an extension that backports the functionality to Contao.

I agree too though. I wanted to create a back-port extension anyway, if this didn't make it into Symfony or Contao 4.13 🙃

**Review by m-vo — APPROVED** [review]
Very nice. :heart: 

Could we maybe add something like a `--test` option that takes the recipient and otherwise uses some opinionated defaults (or even uses the system admin email default for the recipient as well).

 I often find myself needing this kind of quick way to test a SMTP config and could not care less about the actual contents. I just want to verify mails can be send. :smile:

**Comment by aschempp** [comment]
> Could we maybe add something like a `--test` option that takes the recipient and otherwise uses some opinionated defaults (or even uses the system admin email default for the recipient as well).
> 
> I often find myself needing this kind of quick way to test a SMTP config and could not care less about the actual contents. I just want to verify mails can be send. 😄

I understand that requirement, but I think it contradicts with the fact that we want to drop this again as soon as it is supported in Symfony?

**Comment by m-vo** [comment]
> as soon as it is supported in Symfony

which could be anywhere between "in quite a while" and "never" :smile:

**Review by ausi — APPROVED** [review]

**Comment by aschempp** [comment]
As discussed in Slack (and agreed to by most core devs), this might as well be an external bundle. The advantages would be that it can be installed in versions prior to 4.13, it could be kept up-to-date with the latest Symfony PR features (without being related to a Contao version) and we could still require it in a Contao Managed Edition by default (until if Symfony ships this feature).

**Comment by fritzmg** [comment]
Do we want it to be part of `contao` (not the mono repo though)?

**Comment by aschempp** [comment]
> Do we want it to be part of `contao` (not the mono repo though)?

I don't see a reason to. It could be @fritzmg same as we have bundles from @terminal42 and @ausi. Being in the `contao` organisation has no benefits except enforcing it to be maintained by the core team, possibly even after we drop it from Contao.

**Comment by fenepedia** [comment]
@fritzmg Will this feature be shipped with Contao 5?

**Comment by fritzmg** [comment]
@fenepedia see the discussion above - we decided to not ship it with Contao. We might create a `inspiredminds/symfony-mailer-send-command-bundle`.

**Comment by leofeyer** [comment]
Maybe we should close the PR then?

**Comment by fritzmg** [comment]
You can now install [`inspiredminds/contao-mailer-command`](https://github.com/inspiredminds/contao-mailer-command) in the Contao Managed Edition or [`inspiredminds/symfony-mailer-command-bundle`](https://github.com/inspiredminds/symfony-mailer-command-bundle) for standard Symfony projects in order to get a basic `mailer:send` command.

**Comment by aschempp** [comment]
Should we make [inspiredminds/contao-mailer-command](https://github.com/inspiredminds/contao-mailer-command) a default in Contao (5)? Would also allow to implement https://github.com/contao/contao-manager/issues/254

**Comment by leofeyer** [comment]
We can use `mailer:send` from Symfony 6.1 in Contao 5.

**Comment by Toflar** [comment]
Where did you read that Symfony 6.1 provides this command? @fritzmg's PR https://github.com/symfony/symfony/pull/43687 is still open.

**Comment by leofeyer** [comment]
You are right, the PR is now targeted on Symfony 6.2.

**Pull request closed without merging**
